src/analysis/normed_space/riesz_lemma.lean

```diff
@@ -0,0 +1,30 @@
+import analysis.normed_space.basic
+import topology.metric_space.hausdorff_distance
+
+variables {k : Type*} [normed_field k]
+variables {α : Type*} [normed_group α] [normed_space k α]
+
+/-- Riesz's Lemma. Stated in terms of multiples of norms since in
+    general the existence of an element of norm exactly 1 is not
+    guaranteed. -/
+lemma riesz_lemma {β : subspace k α} (hβc : is_closed β.carrier)
+(hβ : ∃ a : α, a ∉ β)
+{r : ℝ} (hr : r < 1) : ∃ x : α, ∀ y : β, r * ∥x∥ ≤ ∥x - y∥ :=
+or.cases_on (le_or_lt r 0)
+(λ hle, ⟨0, λ b, by {rw [norm_zero, mul_zero], exact norm_nonneg _}⟩)
```

Comment:
It seems that Riesz lemma without `x≠0` can be proved by this term without using `r`, `hβc`, `hβ`.

Reply:
I'm sorry, but I don't think I understand what you mean. That term is the proof for when `r < 0`, where the inequality holds trivially, so it makes sense that it invokes none of the other hypotheses. I thought it'd be nice if we could invoke the lemma without supplying the spurious requirement that `r` should be positive.

Or did you mean to point out that there's a nicer way to do the proof without the `cases_on`?

Reply:
oh, wait, nevermind that, I was being a goof. editing in the corrections now.